### PR TITLE
feat(kotlin): Add Unsigned Primitive Support

### PR DESCRIPTION
**Pull request by wywen**
## What does this PR do?

This PR adds unsigned primitive support to Kotlin Fury.
It also adds tests for the standard kotlin primitives(supported by fury Java), nullable primitive tests, boundary tests for unsigned serializers.

## Related issues
#683

## Does this PR introduce any user-facing change?

Yes it adds Unsigned support for Kotlin. There's documentation new issue (should add something to document Kotlin!)

- [ ] Does this PR introduce any public API change?
- [ ] Does this PR introduce any binary protocol compatibility change?

## Benchmark
N/A

### CODE REVIEW

File: kotlin/src/main/kotlin/org/apache/fury/serializer/kotlin/UnsignedSerializer.kt [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    fury: Fury,
+) : Serializers.CrossLanguageCompatibleSerializer<UByte>(
+    fury,
+    UByte::class.java,
+    Type.UINT8.id,
+    fury.isBasicTypesRefIgnored,
+    true) {
+
+    override fun write(buffer: MemoryBuffer, value: UByte) {
+        buffer.writeVarUint32Small7(value.toInt())
```
Comment (chaokunyang) [inline kotlin/src/main/kotlin/org/apache/fury/serializer/kotlin/UnsignedSerializer.kt:43]:
```suggestion
        buffer.writeByte(value.toInt())
```

File: kotlin/src/main/kotlin/org/apache/fury/serializer/kotlin/UnsignedSerializer.kt [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    }
+
+    override fun read(buffer: MemoryBuffer): UByte {
+        return buffer.readVarUint32().toUByte()
```
Comment (chaokunyang) [inline kotlin/src/main/kotlin/org/apache/fury/serializer/kotlin/UnsignedSerializer.kt:47]:
```suggestion
        return buffer.readByte().toUByte()
```

File: kotlin/src/main/kotlin/org/apache/fury/serializer/kotlin/UnsignedSerializer.kt [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    fury: Fury,
+) : Serializers.CrossLanguageCompatibleSerializer<UShort>(
+    fury,
+    UShort::class.java,
+    Type.UINT16.id,
+    fury.isBasicTypesRefIgnored,
+    true
+) {
+    override fun write(buffer: MemoryBuffer, value: UShort) {
+        buffer.writeVarUint32(value.toInt())
```
Comment (chaokunyang) [inline kotlin/src/main/kotlin/org/apache/fury/serializer/kotlin/UnsignedSerializer.kt:68]:
```suggestion
        buffer.writeVarUint16(value.toInt())
```

File: kotlin/src/main/kotlin/org/apache/fury/serializer/kotlin/UnsignedSerializer.kt [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    }
+    override fun read(buffer: MemoryBuffer): UShort {
+        return buffer.readVarUint32().toUShort()
```
Comment (chaokunyang) [inline kotlin/src/main/kotlin/org/apache/fury/serializer/kotlin/UnsignedSerializer.kt:71]:
```suggestion
        return buffer.readVarUint16().toUShort()
```

**Review by chaokunyang — APPROVED** [review]
Looks great

**Comment by chaokunyang** [comment]
> writeVarUint16

My bad, MemoryBuffer didn't provide read/writeVarUint16 method, we can keep using read/writeVarUint32, the performance will be same

**Comment by wywen** [comment]
> > writeVarUint16
> 
> My bad, MemoryBuffer didn't provide read/writeVarUint16 method, we can keep using read/writeVarUint32, the performance will be same

Will doing a reinterpret_cast to short be more compact (kotlin toXYZ methods are `reinterpret_cast` not `static_cast` )? I tried it and it passes boundary and roundtrip tests,

File: kotlin/README.md [hunk 5]
```diff
@@ -9,7 +9,7 @@ Fury Kotlin provides additional tests and implementation support for Kotlin type
 Fury Kotlin is tested and works with the following types:
 
 - primitives: `Byte`, `Boolean`, `Int`, `Short`, `Long`, `Char`, `Float`, `Double`, `UByte`, `UShort`, `UInt`, `UShort`, `UInt`.
--  `Byte`, `Boolean`, `Int`, `Short`, `Long`, `Char`, `Float`, `Double` works out of the box with the default java implementation.
+- `Byte`, `Boolean`, `Int`, `Short`, `Long`, `Char`, `Float`, `Double` works out of the box with the default java implementation.
```
Comment (chaokunyang) [inline kotlin/README.md:12]: Could we replace all `default java implementation` with `default fury java implementation`

**Comment by wywen** [comment]
looks like read/writeUint32 will pick the right representation.
